pkg/ingester/label_names_and_values.go

```diff
@@ -5,14 +5,25 @@ package ingester
 import (
 	"context"
 
+	"github.com/grafana/dskit/concurrency"
 	"github.com/prometheus/prometheus/model/labels"
 	"github.com/prometheus/prometheus/tsdb"
 	"github.com/prometheus/prometheus/tsdb/index"
 
 	"github.com/grafana/mimir/pkg/ingester/client"
 )
 
-const checkContextErrorSeriesCount = 1000 // series count interval in which context cancellation must be checked.
+const (
+	checkContextErrorSeriesCount = 1000 // series count interval in which context cancellation must be checked.
+
+	maxConcurrency = 16
```

Comment:
I would move this constant to `computeLabelValuesSeriesCount`, as otherwise it's available for all the ingester package, which may have many concurrent things happening.